probe/proc/cache.go

```diff
@@ -0,0 +1,139 @@
+package proc
+
+import (
+	"bytes"
+	"errors"
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+const (
+	filesCacheLen        = 1024
+	filesCacheExpiration = 60 * time.Second
+)
+
+var (
+	errInvalidCacheEntry = errors.New("invalid cache entry")
+)
+
+type filesCacheEntry struct {
+	file     File
```

Comment:
Whats the justification for keeping the fds open?  Do you have a trace that shows its saves a lot of CPU?  Could we just cache file contents for certain files without the fds?

I'm not convinced its safe to keep files open in proc for long periods.  There little mention of it on the web (http://stackoverflow.com/questions/1623731/open-close-strategy-for-proc-pseudo-file), but unless we can find something that says its safe (ie the kernel source) I'd prefer not to.